### ISSUE TITLE
chapel build error on macOS big sur

### ISSUE THREAD

**Issue by Mashijams** [chapel-lang/chapel#18600]
### Summary of Problem

I installed chapel through Homebrew as described in "Installing chapel" documentation, but I wasn't able to compile any chapel programs.

The compiler message I got was :
```
  In file included from /usr/local/Cellar/chapel/1.25.0/libexec/runtime/etc/rtmain.c:24:
In file included from /usr/local/Cellar/chapel/1.25.0/libexec/runtime/include/./stdchpl.h:39:
In file included from /usr/local/Cellar/chapel/1.25.0/libexec/runtime/include/./arg.h:24:
In file included from /usr/local/Cellar/chapel/1.25.0/libexec/runtime/include/./chpltypes.h:334:
In file included from /usr/local/Cellar/chapel/1.25.0/libexec/runtime/include/./chpl-string-support.h:35:
/usr/local/Cellar/chapel/1.25.0/libexec/runtime/include/./error.h:26:10: fatal error: 'stdarg.h' file not found
#include <stdarg.h>
         ^~~~~~~~~~
1 error generated.
error: error running clang during code generation
```
### My working to fix it

I thought there are some problems with my command-line tools so I tried to reinstall command-line tools but the issue was still not resolved.

My copy of `stdarg.h` is located at `/library/developer/commandlinetools/usr/lib/clang/13.0.0/include` so I came to conclusion that command-line tools isn't causing trouble.

I began experimenting to fix issue, I copied include files which compiler was not able to find to `libexec/runtime/include` on chapel directory.
Every time I got new missing include file error I simply copied that include file to above chapel directory, until a new issue came :
```
 [Fail] Test job failed to compile - Chapel is not installed correctly
[Fail] Compilation output:
sh: /usr/local/Cellar/llvm@11/11.1.0_2/bin/clang++: No such file or directory
error: Make Binary - Linking
```
I already had `llvm@11` located at `/usr/local/cellar/llvm@11` but the version folder I had was `11.1.0_3` instead of `11.1.0_2`, so I renamed folder to `11.1.0_2` and removed all those extra include files I copied to `libexec/runtime/include` which solved the issue for me.
```
 [Info] $CHPL_LAUNCHER=none is compatible with test script.
[Info] Running test job.
[Info] Test job complete.
SUCCESS: 'make check' passed!

```
In summary, chapel compiler searching for an invalid path in my system might be the cause of this issue.

**Comment by bradcray** [comment]
@Mashijams : Thanks for filing this.  Soon after you reported this on gitter, we had other users and developers start to run into it as well (unfortunately), so appreciate you giving us the heads-up.  I've reproduced it on my laptop as well now that I've upgraded LLVM.

As a workaround for this, I think you should be able to throw the `--target-compiler=clang` flag, which stops relying on LLVM and seems to be working for me.  Let me know if you find otherwise.

@carlocab: I suspect we're going to need to do a 1.25.1 release of Chapel to incorporate the fix Michael mentions in #18551 (both for this, and other related, reasons).  If I were trying to update the current formula to make it resilient to the issue, I'd probably have it build with `CHPL_LLVM=none` to just disable the LLVM back-end and avoid this code temporarily.  Tagging @mppf to see what he thinks about that given his familiarity with the issue and help putting the 1.25.0 homebrew formula together.

@carlocab: I'm not understanding why the `mv` in your example above is failing.  Is it clear to you?

[also tagging @Maxrimus and @tzinsky due to the relation to the homebrew release]

**Comment by bradcray** [comment]
Chapel 1.25.1 contained #18551, so I'm closing this as proposed in the previous comment.  If anyone runs into it in the future, please let us know!
